workflow/controller/operator.go

```diff
@@ -101,7 +101,7 @@ type wfOperationCtx struct {
 	eventRecorder    record.EventRecorder
 	// preExecutionNodePhases contains the phases of all the nodes before the current operation. Necessary to infer
 	// changes in phase for metric emission
-	preExecutionNodePhases map[string]wfv1.NodePhase
+	preExecutionNodePhases map[string]wfv1.NodeStatus
```

Comment:
Is the plan to consolidate the `NodeStatus` and `NodePhase` types into `NodeStatus`? Also, it's a nit, but it would be good to update `preExecutionNodePhases` to `preExecutionNodeStatuses` along with the associated comment

Reply:
Well the nodestatus does contain the node phase inside them, but I agree I should rename this. its mostly used for phases so thats why I didn't bother.

Reply:
I think it should be renamed for sanity